app/src/main/java/com/example/triangleapp/MainActivity.kt

```diff
@@ -1,11 +1,224 @@
+                {
+                    calculateButton.setEnabled(false)
+                }
+                else
+                {
+                    calculateButton.setEnabled(true)
+                }
+            }
+
+            override fun onTextChanged(s: CharSequence, start: Int,
+                                       before: Int, count: Int) {
+                if (s.toString().trim({ it <= ' ' }).isEmpty())
+                {
+                    calculateButton.setEnabled(false)
+                }
+                else
+                {
+                    calculateButton.setEnabled(true)
+                }
+            }
+        })
+
+
+        side2Length.addTextChangedListener(object: TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
+                if (s.toString().isNotEmpty()) {
+                    side2Len = side2Length.text.toString().toInt();
+                }
+            }
+            override fun beforeTextChanged(s: CharSequence, start: Int,
+                                           count: Int, after: Int) {
+                if (s.toString().trim({ it <= ' ' }).isEmpty())
+                {
+                    calculateButton.setEnabled(false)
+                }
+                else
+                {
+                    calculateButton.setEnabled(true)
+                }
+            }
+
+            override fun onTextChanged(s: CharSequence, start: Int,
+                                       before: Int, count: Int) {
+                if (s.toString().trim({ it <= ' ' }).isEmpty())
+                {
+                    calculateButton.setEnabled(false)
+                }
+                else
+                {
+                    calculateButton.setEnabled(true)
+                }
+            }
+        })
+
+        side3Length.addTextChangedListener(object: TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
+                if (s.toString().isNotEmpty()) {
+                    side3Len = side3Length.text.toString().toInt();
+                }
+            }
+            override fun beforeTextChanged(s: CharSequence, start: Int,
+                                           count: Int, after: Int) {
+                if (s.toString().trim({ it <= ' ' }).isEmpty())
+                {
+                    calculateButton.setEnabled(false)
+                }
+                else
+                {
+                    calculateButton.setEnabled(true)
+                }
+            }
+
+            override fun onTextChanged(s: CharSequence, start: Int,
+                                       before: Int, count: Int) {
+                if (s.toString().trim({ it <= ' ' }).isEmpty())
+                {
+                    calculateButton.setEnabled(false)
+                }
+                else
+                {
+                    calculateButton.setEnabled(true)
+                }
+            }
+        })
```

Comment:
Can we wrap those functions as private member functions to pass side1, 2, and 3? So, we don't need to repeat the code 3 times?